src/main/java/controller/LottoGame.java

```diff
@@ -0,0 +1,40 @@
+package controller;
+
+import model.*;
+import view.InputView;
+import view.OutputView;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+public class LottoGame {
+
+    public static void main(String[] args) {
+        Money money = new Money(InputView.inputMoneyAmount());
+        Lottos lottos = Lottos.issueLottos(money, new AutoLottoGenerator());
+        OutputView.printLottos(lottos);
+
+        WinningLotto winningLotto = inputWinningLotto();
+        LottoStatistics lottoStatistics = new LottoStatistics(lottos.produceStatistics(winningLotto));
+        double yield = lottoStatistics.calculateYield(money);
+        OutputView.printStatistics(lottoStatistics.getStatistics(), yield);
+    }
+
+    private static WinningLotto inputWinningLotto() {
+        Lotto lotto = new Lotto(convertNumbersToLottoNumbers(InputView.inputNumbers()));
+        LottoNumber bonusNumber = convertNumberToLottoNumber(InputView.inputNumber());
+        return new WinningLotto(lotto, bonusNumber);
+    }
+
+    private static List<LottoNumber> convertNumbersToLottoNumbers(List<Integer> numbers) {
+        return numbers.stream()
+                .map(LottoNumber::new)
+                .collect(toList());
+    }
+
+    private static LottoNumber convertNumberToLottoNumber(int number) {
+        return new LottoNumber(number);
```

Comment:
마찬가지로 캐싱을 했는데 막상 그 값을 사용하고 있지는 않네요. 
이것이 생성자를 public으로 뒀기 때문이라고 볼 수 있겠죠? 😉
```suggestion
        return LottoNumber.getLottoNumberFromCache(number);
```

Reply:
감사합니다!